app/src/main/java/com/hover/stax/database/DatabaseRepo.kt

```diff
@@ -0,0 +1,318 @@
+        get() = transactionDao.transactionsForAppReview
+
+    fun getSchedule(id: Int): Schedule {
+        return scheduleDao[id]
+    }
+
+    fun insert(schedule: Schedule?) {
+        AppDatabase.databaseWriteExecutor.execute { scheduleDao.insert(schedule) }
+    }
+
+    fun update(schedule: Schedule?) {
+        AppDatabase.databaseWriteExecutor.execute { scheduleDao.update(schedule) }
+    }
+
+    fun delete(schedule: Schedule?) {
+        AppDatabase.databaseWriteExecutor.execute { scheduleDao.delete(schedule) }
+    }
+
+    // Requests
+    val liveRequests: LiveData<List<Request>>
+        get() = requestDao.liveUnmatched
+
+    fun getLiveRequests(channelId: Int): LiveData<List<Request>> {
+        return requestDao.getLiveUnmatchedByChannel(channelId)
+    }
+
+    val requests: List<Request>
+        get() = requestDao.unmatched
+
+    fun getRequest(id: Int): Request {
+        return requestDao[id]
+    }
+
+    fun decrypt(encrypted: String, c: Context): LiveData<Request> {
+        decryptedRequest.value = null
+        val removedBaseUrlString = encrypted.replace(c.getString(R.string.payment_root_url, ""), "")
+
+        //Only old stax versions contains ( in the link
+        if (removedBaseUrlString.contains("(")) decryptRequestForOldVersions(removedBaseUrlString)
+        else decryptRequest(removedBaseUrlString, c)
+        return decryptedRequest
+    }
+
+    private fun decryptRequest(param: String, c: Context) {
+        decryptedRequest.postValue(Request(Request.decryptBijective(param, c)))
+    }
+
+    private fun decryptRequestForOldVersions(param: String) {
+        var params = param
+        try {
+            val e = Request.getEncryptionSettings().build()
+            if (Request.isShortLink(params)) {
+                params = Shortlink(params).expand()
+            }
+            e.decryptAsync(params.replace("[(]".toRegex(), "+"), object : Encryption.Callback {
+                override fun onSuccess(result: String) {
+                    decryptedRequest.postValue(Request(result))
+                }
+
+                override fun onError(exception: Exception) {
+                    Utils.logErrorAndReportToFirebase(TAG, "failed link decryption", exception)
+                }
+            })
+        } catch (e: NoSuchAlgorithmException) {
+            Utils.logErrorAndReportToFirebase(TAG, "decryption failure", e)
+        }
+    }
+
+    fun insert(request: Request?) {
+        AppDatabase.databaseWriteExecutor.execute { requestDao.insert(request) }
+    }
+
+    fun update(request: Request?) {
+        AppDatabase.databaseWriteExecutor.execute { requestDao.update(request) }
+    }
+
+    fun delete(request: Request?) {
+        AppDatabase.databaseWriteExecutor.execute { requestDao.delete(request) }
+    }
+
+    companion object {
+        private const val TAG = "DatabaseRepo"
+    }
+
+    init {
+        val db = AppDatabase.getInstance(application)
```

Comment:
The app and sdk databases can be passed as constructor arguments in the class declaration. You can then modify the DatabaseRepo instantiation in the `module.kt` file to make them injectable.

Reply:
I tried this, but this will require refactoring all ViewModels, which was causing more errors, and I'm unable to test because certain bugs in this branch aren't resolved here, instead resolve in the base branch.

Easier to do the refractoring on development.